controllers/storagecluster/rookoperatorhostnetwork.go

```diff
@@ -0,0 +1,104 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type rookCephCsvHostNetwork struct{}
+
+// if the StorageCluster is configured to run on non-default host network, rook-ceph-operator needs to run on host network as well
+// this is validated by checking if the AddressRanges.Public field is set in the Network spec
+// since pod network cannot always communicate with non-default host network rook-ceph-operator needs to run on host network
+func (obj *rookCephCsvHostNetwork) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if shouldUseHostNetworking(instance) {
+		err := updateOpHostNetworkInRookCephCSV(r.ctx, r.Client, instance.Namespace, true)
+		if err != nil {
+			r.Log.Error(err, "Failed to update Rook Ceph Operator HostNetwork.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
+			return reconcile.Result{}, err
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func (obj *rookCephCsvHostNetwork) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	err := updateOpHostNetworkInRookCephCSV(r.ctx, r.Client, instance.Namespace, false)
+	if err != nil {
+		r.Log.Error(err, "Failed to update Rook Ceph Operator HostNetwork.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
```

Comment:
Why do we need to revert anything here when the storage cluster is getting deleted? 
Wouldn't a new storage cluster will just re-set (using ensure created) to the proper value when it will get created?

Reply:
CSV changes will not be removed once ensure created changes it (unless ensureCreated checks the rook-csv every time and set appropriate host nw value)